src/cpp/session/modules/build/SessionBuild.cpp

```diff
@@ -1157,7 +1157,7 @@ class Build : boost::noncopyable,
       std::vector<std::string> rSourceCommands;
       
       boost::format fmt(
-         "if (nzchar('%1%')) devtools::load_all('%1%');"
+         "if (nzchar('%1%')) devtools::load_all(devtools::as.package(dirname('%2%')));"
```

Comment:
If I understand correctly, this works because `devtools::as.package()` will search parent directories for the package root?

Reply:
Right, that is correct!